gravitee-am-gateway/gravitee-am-gateway-handler/gravitee-am-gateway-handler-core/src/main/java/io/gravitee/am/gateway/handler/vertx/VertxSecurityDomainHandler.java

```diff
@@ -98,13 +102,28 @@ protected void doStart() throws Exception {
     protected void doStop() throws Exception {
         logger.info("Security domain [" + domain.getName() + "] handler is now stopping, closing context...");
 
+
         stopComponents();
         stopProtocols();
 
         super.doStop();
+
+        stopApplicationContext();
+
         logger.info("Security domain [" + domain.getName() + "] handler is now stopped", domain);
     }
 
+    private void stopApplicationContext() {
+        if (applicationContext instanceof ConfigurableApplicationContext) {
+            try {
```

Comment:
Unless it is targeted to be retrofitted in older version you can use :
`applicationContext instanceof ConfigurableApplicationContext confContext` to avoid redundant cast